src/dbos-executor.ts

```diff
@@ -372,6 +371,15 @@ export class DBOSExecutor {
       } finally {
         this.tracer.endSpan(wCtxt.span);
       }
+      // Asynchronously flush the result buffer.
+      const resultBufferFlush: Promise<void> = this.userDatabase.transaction(async (client: UserDatabaseClient) => {
```

Comment:
I'm only worried if we have many many temp read-only point read transaction workflows, we will add too much write load to Postgres. Another way to implement this is to make the result buffer a global one, which allows us to batch insert to PG.

Reply:
That's a good future optimization if this becomes a problem